pkg/rpcclient/actor/actor.go

```diff
@@ -285,3 +287,17 @@ func (a *Actor) SendUncheckedRun(script []byte, sysfee int64, attrs []transactio
 func (a *Actor) Sender() util.Uint160 {
 	return a.txSigners[0].Account
 }
+
+// WaitSuccess is similar to [waiter.Wait], but also checks for the VM state
+// to be HALT (successful execution). Execution result is still returned in
+// case you need to examine events or stack.
+func (a *Actor) WaitSuccess(h util.Uint256, vub uint32, err error) (*state.AppExecResult, error) {
+	aer, err := a.Wait(h, vub, err)
+	if err != nil {
+		return nil, err
+	}
+	if aer.VMState != vmstate.Halt {
+		return nil, fmt.Errorf("vm failed: %s", aer.FaultException)
```

Comment:
Don't we need a designated error so that the user is able to make `errors.Is` and distinguish this error from some other?

Reply:
Had this thought, but I don't see it being useful at this stage. Exporting errors is a good practice in general though. Theoretically, we may need to separate "transaction accepted", but failed and "transaction never went in" cases. `waiter.ErrTxNotAccepted` is about that. OK, added.